Src/AutoFixture/DataAnnotations/RangedRequest.cs

```diff
@@ -133,8 +135,28 @@ public override string ToString()
                 "Workaround for a bug in CA: https://connect.microsoft.com/VisualStudio/feedback/details/521030/")]
         private static object GetConvertedRangeBoundary(object attributeValue, Type conversionType)
         {
+            Debug.Assert(attributeValue != null, "Value cannot be null as it's asserted in constructor.");
```

Comment:
What does the [Zen of Python](https://www.python.org/dev/peps/pep-0020/) (explicit is better than implicit) tell us here?In the rest of the codebase we use [Guard Clauses](http://wiki.c2.com/?GuardClause) in the form of `if cond then exp`... 😕

Reply:
Well, I had a special reasoning about this decision 😉

I always use Guard Clauses to verify the incoming _public_ data, which is untrusted and potentially could be anything. 

However in our case the API is private and we totally control the data we pass to this method. Logically, the value can be _never_ null, because it has been already verified. However, that might be not obvious (especially to R#) that `null` can be never passed to this method. That's why I leave the `Debug.Assert()` - they are not compiled and are used just to indicate some internal contract we have.

I prefer to use the Guard Clauses to verify the public non-trusted data, while use `Debug.Assert` to indicate some internal contract that could never fail (until you change the implementation, so it's broken). Usually I use [R# Code Annotations](https://www.jetbrains.com/help/resharper/Code_Analysis__Code_Annotations.html) to indicate such things so such assertions are not even needed, but here they are missing. Probably, later we could rely on [non-nullable references](https://blogs.msdn.microsoft.com/dotnet/2017/11/15/nullable-reference-types-in-csharp/) when they appear in C#.

Does this reasoning make sense to you, or you still persist on Guard Clause and don't want to distinguish between public and internal contact? ☺️

Reply:
I've never found `Debug.Assert` useful. That being said, I wouldn't use it also here. However, I think we have to draw a line; does it matter if I (dis)like it? *You* are doing all the hard work, after all...

Reply:
Well, I don't know 😅 I use `Debug` to assert a thing which could _never_ happen. On other hand, regular assertions describe a _potentially_ occurring thing, which is not expected. I prefer to distinguish between those cases. Therefore, if you don't strongly persist, I'd keep that as is.

It's not critically important to me, so I'll keep that up to you 😉

Reply:
Well, it's up to _you_. Personally, I never use it, and never found it useful. I just use `if-else` statements instead.

Reply:
Rewrote to a regular Null Guard, so we can proceed and merge this 😉

Reply:
I said it's up to _you_ though 🤣 🥇

Reply:
Yep, but you _haven't approved_ the PR, meaning that you expected some actions from my side 😉 Given that this discussion already took much time, I decided that change is not so important 😄